src/Sarif/Autogenerated/RunEqualityComparer.cs

```diff
@@ -0,0 +1,339 @@
+                }
+
+                if (obj.Files != null)
+                {
+                    // Use xor for dictionaries to be order-independent.
+                    int xor_0 = 0;
+                    foreach (var value_6 in obj.Files)
+                    {
+                        xor_0 ^= value_6.Key.GetHashCode();
+                        if (value_6.Value != null)
+                        {
+                            xor_0 ^= value_6.Value.GetHashCode();
+                        }
+                    }
+
+                    result = (result * 31) + xor_0;
+                }
+
+                if (obj.LogicalLocations != null)
+                {
+                    // Use xor for dictionaries to be order-independent.
+                    int xor_1 = 0;
+                    foreach (var value_7 in obj.LogicalLocations)
+                    {
+                        xor_1 ^= value_7.Key.GetHashCode();
+                        if (value_7.Value != null)
+                        {
+                            xor_1 ^= value_7.Value.GetHashCode();
+                        }
+                    }
+
+                    result = (result * 31) + xor_1;
+                }
+
+                if (obj.Results != null)
+                {
+                    foreach (var value_8 in obj.Results)
+                    {
+                        result = result * 31;
+                        if (value_8 != null)
+                        {
+                            result = (result * 31) + value_8.GetHashCode();
+                        }
+                    }
+                }
+
+                if (obj.ToolNotifications != null)
+                {
+                    foreach (var value_9 in obj.ToolNotifications)
+                    {
+                        result = result * 31;
+                        if (value_9 != null)
+                        {
+                            result = (result * 31) + value_9.GetHashCode();
+                        }
+                    }
+                }
+
+                if (obj.ConfigurationNotifications != null)
+                {
+                    foreach (var value_10 in obj.ConfigurationNotifications)
+                    {
+                        result = result * 31;
+                        if (value_10 != null)
+                        {
+                            result = (result * 31) + value_10.GetHashCode();
+                        }
+                    }
+                }
+
+                if (obj.Rules != null)
+                {
+                    // Use xor for dictionaries to be order-independent.
+                    int xor_2 = 0;
+                    foreach (var value_11 in obj.Rules)
+                    {
+                        xor_2 ^= value_11.Key.GetHashCode();
+                        if (value_11.Value != null)
+                        {
+                            xor_2 ^= value_11.Value.GetHashCode();
```

Comment:
We shouldn't be calling the default Sarif object GetHashCode here but the one from the associated equality comparer, right?